### PR TITLE
[MM-27207] e2e archived channels I (3): [MM-T1676] CTRL/CMD+K does not show private archived channels you are not a member of

### PR DESCRIPTION
**Pull request by Willyfrog**
#### Summary

[MM-T1676] CTRL/CMD+K does not show private archived channels you are not a member of

#### Ticket Link
[MM-27207](https://mattermost.atlassian.net/browse/MM-27207)

Test:
- [MM-T1676](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin%3Acom.kanoah.test-manager__main-project-page#!/testCase/MM-T1676)

#### Related Pull Requests
Requires:

https://github.com/mattermost/mattermost-webapp/pull/6362

https://github.com/mattermost/mattermost-webapp/pull/6358

### CODE REVIEW

File: e2e/cypress/integration/channel/archived_channels_spec.js [hunk 1]
```diff
@@ -55,5 +67,204 @@ describe('Leave an archived channel', () => {
...
+                name: 'archived-',
+                type: 'P',
+                teamId: testTeam.id,
+                teamName: testTeam.name,
+            },
+            [`some text message ${getRandomId()}`],
+        ).then(() => {
+            cy.visit(`/${testTeam.name}/channels/off-topic`);
+
+            // Select CTRL/⌘+k) to open the channel switcher
```
Comment (srkgupta) [inline e2e/cypress/integration/channel/archived_channels_spec.js:211]: Suggested change:

`// # Select`

File: e2e/cypress/integration/channel/archived_channels_spec.js [hunk 2]
```diff
@@ -55,5 +67,204 @@ describe('Leave an archived channel', () => {
...
+            cy.typeCmdOrCtrl().type('K', {release: true});
+
+            // Start typing the name of a private channel on this team that has been archived which the test user belongs to
```
Comment (srkgupta) [inline e2e/cypress/integration/channel/archived_channels_spec.js:214]: Suggested change:

`// # Start typing....`

File: e2e/cypress/integration/channel/archived_channels_spec.js [hunk 3]
```diff
@@ -55,5 +67,204 @@ describe('Leave an archived channel', () => {
...
+            cy.get('#suggestionList').find('.icon-globe').should('be.visible');
+        });
+    });
+    it('MM-T1675 CTRL/CMD+K list private archived channels you are a member of', () => {
+        cy.visit(`/${testTeam.name}/channels/off-topic`);
+
+        // Select CTRL/⌘+k) to open the channel switcher
+        cy.typeCmdOrCtrl().type('K', {release: true});
+
+        // Start typing the name of a private channel on this team that has been archived which the test user belongs to
```
Comment (srkgupta) [inline e2e/cypress/integration/channel/archived_channels_spec.js:231]: Comments should start with `#` for actions

File: e2e/cypress/integration/channel/archived_channels_spec.js [hunk 4]
```diff
@@ -55,5 +67,204 @@ describe('Leave an archived channel', () => {
...
+        cy.apiLogin(testUser);
+        cy.visit(`/${testTeam.name}/channels/off-topic`);
+        cy.typeCmdOrCtrl().type('K', {release: true});
+
+        // # Start typing the name of a private channel located above
+        cy.get('#quickSwitchInput').type('archived-');
+        cy.wait(TIMEOUTS.HALF_SEC);
+
+        // * Private archived channels you are not a member above are not available on channel switcher
+        cy.contains('#suggestionList', 'archived-not-mine').should('not.exist');
```
Comment (srkgupta) [inline e2e/cypress/integration/channel/archived_channels_spec.js:268]: Suggested change:

`cy.contains('#suggestionList', otherChannelName).should('not.exist');`

**Review by srkgupta — CHANGES_REQUESTED** [review]
Hi @Willyfrog 

Great Work. Overall the test looks great and is working fine. However I have some minor comments suggested. Please do the needful.

**Review by srkgupta — APPROVED** [review]
Thanks @Willyfrog. The test looks great and is working fine. Approving the PR.

**Review by hmhealey — APPROVED** [review]

**Pull request merged**